NDTensors/src/dense/densetensor.jl

```diff
@@ -199,7 +199,7 @@ function permutedims!(
 ) where {N,StoreT<:StridedArray}
   RA = array(R)
   TA = array(T)
-  RA .= permutedims!(leaf_parenttype(RA), RA, leaf_parenttype(TA), TA, perm)
+  RA = permutedims!(leaf_parenttype(RA), RA, leaf_parenttype(TA), TA, perm)
```

Comment:
Another case where we should call the version without `leaf_parenttype`. Also, we shouldn't assign to `RA` with `RA = ` here since `permutedims!` is just in-place. Unlike the maybe-permute version `permutedims!!`, we shouldn't assume all in-place functions will output the object being modified.